### DISCUSSION TITLE
Matlab toolbox equivalent functions

### DISCUSSION THREAD

**Discussion by zdreams** [petercorke/robotics-toolbox-python#400] [Q&A] [answered]
Hello. 
I am trying to map a robot manipulator's trajectory. Using Matlab toolbox I can use something like this (taken from the official blog):
T0 = trvec2tform([0 0 0]);
Tf = trvec2tform([1 2 3])*eul2tform([pi/2 0 pi/4],'ZYX');
tTimes = linspace(0,1,51);
tInterval = [0 5];
[s,sd,sdd] = trapveltraj([0 1],numel(tTimes));
[T,dT,ddT] = transformtraj(T0,Tf,tInterval,tTimes,'TimeScaling',[s;sd;sdd]);
plotTransforms(tform2trvec(T),tform2quat(T));
I am unfamiliar the equivalent trajectory functions in the Py toolbox. Thanks

**Comment by vla-gove** [comment, marked as answer]
This has been answered by @petercorke at [Linear Trajectory](https://github.com/petercorke/robotics-toolbox-python/discussions/262):

*  `trapveltraj` maps to `trapezoidal`
* `rottraj` can be handled by the `interp` method of `UnitQuaternion` or `SO3`
* `transformtraj` can be handled by the `interp` method of `SE3` or `Twist3`, or the `ctraj` function.

_Originally posted by @petercorke in https://github.com/petercorke/robotics-toolbox-python/discussions/262#discussioncomment-2960087_
